internal/overlord/servstate/handlers.go

```diff
@@ -75,9 +75,14 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 		return fmt.Errorf("cannot find service %q in plan", req.Name)
 	}
 
-	_, previous := m.services[req.Name]
-	if previous {
-		return fmt.Errorf("service %q was previously started", req.Name)
+	if service, previous := m.services[req.Name]; previous {
+		select {
+		case <-service.done:
+			// Service exited but not yet removed, safe to override.
```

Comment:
This is only safe because we check for the cmd field before actually removing. We might probably give a hint here so future generations don't fall into the trap of thinking that this is always safe, and we can also be more explicit about what it is that it's safe to do. Something along the lines of:
```
// Service exited but not yet removed. It's safe to replace it
// because the removal will check if the cmd still matches.
```